java/flight/src/main/java/org/apache/arrow/flight/FlightService.java

```diff
@@ -191,17 +191,17 @@ public void completed() {
             StreamPipe.wrap(responseObserver, PutResult::toProtocol)).run();
         responseObserver.onCompleted();
       } catch (Exception ex) {
-        logger.error("Failed to process custom put.", ex);
         responseObserver.onError(ex);
-        // The client may have terminated, so the exception here is effectively swallowed.
         // Log the error as well so -something- makes it to the developer.
         logger.error("Exception handling DoPut", ex);
-      }
-      try {
-        fs.close();
-      } catch (Exception e) {
-        logger.error("Exception closing Flight stream", e);
-        throw new RuntimeException(e);
+        throw new RuntimeException(ex);
+      } finally {
```

Comment:
Out of curiosity whatwhat exception was being thrown that we needed the finally?

Reply:
Frankly speaking, I don't know. I am not an expert here. I just think it is a good way to avoid potential resource leak.

It seems @lihalite is an expert here, so he may know more about it.

Reply:
The `finally` makes sure we clean up the `FlightStream` even if the service implementation (`FlightProducer.acceptPut`) throws some exception.